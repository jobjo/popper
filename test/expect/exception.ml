open Popper
open Sample.Syntax

let test_e1 = test @@ fun () -> assert false

let test_e2 =
  test @@ fun () ->
  let* () = Sample.return () in
  assert false

let test_e3 =
  test @@ fun () ->
  let* _ = Sample.int in
  assert false

let test_e4 =
  test @@ fun () ->
  let* n = Sample.int in
  if n mod 2 = 0 then
    fail "Not even"
  else
    assert false

let suite =
  suite
    [ ("Exception 1", test_e1)
    ; ("Exception 2", test_e2)
    ; ("Exception 3", test_e3)
    ; ("Exception 4", test_e4)
    ]

let () = Clean.run suite
