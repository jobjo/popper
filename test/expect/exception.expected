
0/4 tests passed and 4 failed in XXs:

  ✖  Exception 1  Failed
  ✖  Exception 2  Failed
  ✖  Exception 3  Failed
  ✖  Exception 4  Failed after 1 sample

┌────────────────────────────────────────────────────────────────────────────────────────┐
│ Failed `Exception 1'                                                                   │
└────────────────────────────────────────────────────────────────────────────────────────┘

  Reason:

    File "test/expect/exception.ml", line 4, characters 32-38: Assertion failed

┌────────────────────────────────────────────────────────────────────────────────────────┐
│ Failed `Exception 2'                                                                   │
└────────────────────────────────────────────────────────────────────────────────────────┘

  Reason:

    File "test/expect/exception.ml", line 9, characters 2-8: Assertion failed

┌────────────────────────────────────────────────────────────────────────────────────────┐
│ Failed `Exception 3'                                                                   │
└────────────────────────────────────────────────────────────────────────────────────────┘

  Reason:

    File "test/expect/exception.ml", line 14, characters 2-8: Assertion failed

┌────────────────────────────────────────────────────────────────────────────────────────┐
│ Failed `Exception 4' after 1 sample and 2 shrinks.                                     │
└────────────────────────────────────────────────────────────────────────────────────────┘

  Reason:

    File "test/expect/exception.ml", line 22, characters 4-10: Assertion failed

Test Failed: 0/4 tests passed and 4 failed in XXs.

